pkg/controller/expcertificates/issuing/issuing_controller.go

```diff
@@ -267,7 +279,10 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 		return c.issueCertificate(ctx, log, nextRevision, crt, req, pkData)
 	}
 
-	// issueTemporaryCertificate
+	// Issue temporary certificate if needed
+	if err := c.ensureTemporaryCertificate(ctx, crt, pkData); err != nil {
```

Comment:
`ensureTemporaryCertificate` decodes the `pkData` and does not need access to the raw bytes - can we change this to accept a `crypto.Signer` instead, and pass `pk`?

Reply:
It does need the raw data  since it write the private key to the secret

Reply:
I don't think that's correct though - the data stored in the 'next private key secret' could be in a different format to what is specified in the Certificate resource. PKCS8 is used to store the private key in the 'nextPrivateKeySecretName' but the user may be expecting PKCS1 in the target spec.secretName. Doesn't the 'SecretStore' do this automatically for us? I thought we gave it a crypto.Signer/x509.Certificate and it correctly encoded them for us before persisting?

Reply:
What's the status of this? It was my understanding the temporary key is getting stored into the correct format